format function fails (at compile time) if there are fewer arguments than expected

for
`std.debug.warn("hi","not expected");`
there is an explicit error message("unused arguments"), but for
`std.debug.warn("{} {}","hi");`
there is not. instead
```
lib/zig/std/fmt.zig:132:29: error: index 1 outside argument list of size 1
                        args[arg_to_print],
```
is printed.

really minor issue, but if one case is handled, the other one should be as well.

--- a/lib/std/fmt.zig
+++ b/lib/std/fmt.zig
@@ -167,6 +167,10 @@ pub fn format(
                 '}' => {
                     const arg_to_print = comptime nextArg(&used_pos_args, maybe_pos_arg, &next_arg);
 
+                    if (arg_to_print >= args.len) {
+                        @compileError("Too few arguments");
+                    }
+
                     try formatType(
                         args[arg_to_print],
                         fmt[0..0],

